--- /Applications/MAMP/htdocs/head/lib/form/editor_ORIG.php	2015-01-30 14:01:12.000000000 +0100
+++ /Applications/MAMP/htdocs/head/lib/form/editor.php	2015-03-30 18:25:45.000000000 +0200
@@ -270,11 +270,7 @@
      * @return string
      */
     function getElementTemplateType() {
-        if ($this->_flagFrozen){
-            return 'nodisplay';
-        } else {
-            return 'default';
-        }
+        return 'default';
     }
 
     /**
@@ -441,7 +437,9 @@
      * @return empty string
      */
     function getFrozenHtml() {
+        $value = $this->_values['text'];
 
-        return '';
+        return ('' != $value? $value: '&nbsp;')  .
+               $this->_getPersistantData();
     }
 }
